hadoop-ozone/dist/src/main/compose/testlib.sh

```diff
@@ -337,7 +337,26 @@ stop_docker_env(){
   copy_daemon_logs
   save_container_logs
   if [ "${KEEP_RUNNING:-false}" = false ]; then
-     docker-compose --ansi never down
+    down_repeats=3
+    result=0
+    for i in $(seq 1 $down_repeats)
+    do
+      docker-compose --ansi never down
+      result=$?
+      if [[ $result -eq 0 ]]
+      then
+        echo "Docker-compose down successful"
+        break
+      else
+        echo "Error while trying to shut down docker containers"
+        sleep 5
+      fi
```

Comment:
Since `exit-on-error` is enabled (`set -e` at the start of the script), this exits as soon as `docker-compose down` fails.  This can be fixed by running the command as an `if` condition.

Also, I would omit these extra verbose messages, I think output from `docker-compose` is enough.

```suggestion
      if docker-compose --ansi never down; then
        return
      fi
      sleep 5
```